internal/web/server.go

```diff
@@ -1122,3 +1123,35 @@ func isPortConflictError(err error) bool {
 	}
 	return false
 }
+
+func (s *server) repositoryconfig(w http.ResponseWriter, r *http.Request) {
+	switch r.Method {
+	case http.MethodGet:
+		// Handle GET request: Render the repository.html template
+		s.handleGetRepositoryConfig(w, r)
+	case http.MethodPost:
+		// Handle POST request: Update the repository configuration
+		// s.handlePostRepositoryConfig(w, r)
+		fmt.Printf("POST Mehtod Part")
+	default:
+		// Method not allowed
+		http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
+	}
+}
+
+func (s *server) handleGetRepositoryConfig(w http.ResponseWriter, r *http.Request) {
+	var repos v1alpha1.PackageRepositoryList
+	if err := s.pkgClient.PackageRepositories().GetAll(r.Context(), &repos); err != nil {
+		s.respondAlertAndLog(w, err, "Failed to fetch repositories", "danger")
+		return
+	}
+
+	tmplErr := s.templates.settingsPageTmpl.Execute(w, s.enrichPage(r, map[string]any{
```

Comment:
You are rendering the settings page here, but you will have to change this to a new template that you register in `templates.go` (there you will link your `repository.html`).

Reply:
Okay